src/worker/jobs/daily_db_maintenance.rs

```diff
@@ -22,14 +22,14 @@ impl BackgroundJob for DailyDbMaintenance {
     /// archive daily download counts and drop historical data, we can drop this task and rely on
     /// auto-vacuum again.
     async fn run(&self, env: Self::Context) -> anyhow::Result<()> {
-        spawn_blocking(move || {
-            let mut conn = env.connection_pool.get()?;
-
+        let conn = env.deadpool.get().await?;
+        conn.interact(move |conn| {
             info!("Running VACUUM on version_downloads table");
-            sql_query("VACUUM version_downloads;").execute(&mut *conn)?;
+            sql_query("VACUUM version_downloads;").execute(conn)?;
             info!("Finished running VACUUM on version_downloads table");
             Ok(())
         })
         .await
+        .map_err(|err| anyhow!(err.to_string()))?
```

Comment:
It's a little unfortunate we have to launder this through `Display` and lose any semantic information on the error, but looking at `InteractError`, I don't see any obvious way around that.

Reply:
yeah, it's unfortunate. I hope this will resolve itself once we move to `diesel-async` though, since we no longer need the `interact()` calls then